security-admin/src/main/java/org/apache/ranger/security/web/filter/AltiscaleRangerAuthFilter.java

```diff
@@ -0,0 +1,477 @@
+			}
+		};
+		super.init(myConf);
+		LOG.debug("AltiscaleRangerAuthFilter.init(FilterConfig filterConfig) ==> ended");
+	}
+
+	/**
+	 * This function performs post tasks after {@link AuthenticationFilter#doFilter(ServletRequest, ServletResponse, FilterChain)}
+	 * to create a ranger session if the request is authenticated.
+	 */
+	@Override
+	protected void doFilter(FilterChain filterChain, HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
+
+		String userName = null;
+		boolean checkCookie = response.containsHeader("Set-Cookie");
+		if(checkCookie){
+			Collection<String> authUserName = response.getHeaders("Set-Cookie");
+			if(authUserName != null){
+				Iterator<String> i = authUserName.iterator();
+				while(i.hasNext()){
+					String cookie = i.next();
+					if(!StringUtils.isEmpty(cookie)){
+						if(cookie.toLowerCase().startsWith(AuthenticatedURL.AUTH_COOKIE.toLowerCase()) && cookie.contains("u=")){
+							String[] split = cookie.split(";");
+							if(split != null){
+								for(String s : split){
+									if(!StringUtils.isEmpty(s) && s.toLowerCase().startsWith(AuthenticatedURL.AUTH_COOKIE.toLowerCase())){
+										int ustr = s.indexOf("u=");
+										if(ustr != -1){
+											int andStr = s.indexOf("&", ustr);
+											if(andStr != -1){
+												try{
+													userName = s.substring(ustr+2, andStr);
+												}catch(Exception e){
+													userName = null;
+												}
+											}
+										}
+									}
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+		// if security context does not have a user session, authenticate the security context and create a session
+		if(!isAuthenticated()){
+			String defaultUserRole = "ROLE_USER";
+			//if the userName is found on the token then log into ranger using the same user
+			if (userName != null && !userName.trim().isEmpty()) {
+				final List<GrantedAuthority> grantedAuths = new ArrayList<>();
+				grantedAuths.add(new SimpleGrantedAuthority(defaultUserRole));
+				final UserDetails principal = new User(userName, "",grantedAuths);
+				final Authentication finalAuthentication = new UsernamePasswordAuthenticationToken(principal, "", grantedAuths);
+				WebAuthenticationDetails webDetails = new WebAuthenticationDetails(request);
+				((AbstractAuthenticationToken) finalAuthentication).setDetails(webDetails);
+				RangerAuthenticationProvider authenticationProvider = new RangerAuthenticationProvider();
+				authenticationProvider.setAlt_ssoEnabled(true);
+				Authentication authentication = authenticationProvider.authenticate(finalAuthentication);
+				authentication = getGrantedAuthority(authentication);
+				SecurityContextHolder.getContext().setAuthentication(authentication);
+			}
+		}
+		// Delegate call to next filters
+		super.doFilter(filterChain, request, response);
+	}
+
+	private Authentication getGrantedAuthority(Authentication authentication) {
+		UsernamePasswordAuthenticationToken result=null;
```

Comment:
why is temp var result not inside the if statement if you only use it in that scope?